index.html

```diff
@@ -196,6 +196,55 @@
         openmct.install(openmct.plugins.Clock({ enableClockIndicator: true }));
         openmct.install(openmct.plugins.Timer());
         openmct.install(openmct.plugins.Timelist());
+
+        openmct.install(openmct.plugins.FaultManagementPlugin({
+            faultActionProvider: {
+                acknowledgeFault(fault, { comment = '' }) {
+                    console.log('acknowledgeFault', fault);
+                    console.log('comment', comment);
+                },
+                shelveFault(fault, shelveData) {
+                    console.log('shelveFault', fault);
+                    console.log('shelveData', shelveData);
+                }
+            },
+            realtimeFaultProvider: {
+                supportsSubscribe(domainObject) {
+                    return domainObject.type === 'faultManagement';
+                },
+                subscribe(domainObject, callback) {
+                    const faults = JSON.parse(localStorage.getItem('faults') || '{}');
+
+                    function getRandomIndex(start, end) {
+                        return Math.floor(start + (Math.random() * (end - start + 1)));
+                    }
+
+                    let id = setInterval(() => {
+                        const index = getRandomIndex(0, faults.alarms.length - 1);
+                        faults.alarms[index].parameterDetail.currentValue.engValue.doubleValue = Math.random();
```

Comment:
Might want to  look at what limitProviders do here. There's a mapping of yamcs data structure to a predictable Open MCT structure.

Reply:
this is just sample data , can improve it later